cpp/benchmarks/reduction/reduce_benchmark.cpp

```diff
@@ -64,11 +64,12 @@ void BM_reduction(benchmark::State& state, std::unique_ptr<cudf::aggregation> co
   }                                                               \
   BENCHMARK_REGISTER_F(Reduction, name)                           \
     ->UseManualTime()                                             \
-    ->Arg(10000)      /* 10k */                                   \
-    ->Arg(100000)     /* 100k */                                  \
-    ->Arg(1000000)    /* 1M */                                    \
-    ->Arg(10000000)   /* 10M */                                   \
-    ->Arg(100000000); /* 100M */
+    ->Arg(10000)     /* 10k */                                    \
+    ->Arg(100000)    /* 100k */                                   \
+    ->Arg(1000000)   /* 1M */                                     \
+    ->Arg(10000000)  /* 10M */                                    \
+    ->Arg(100000000) /* 100M */                                   \
+    ->Iterations(1000);
```

Comment:
Is this necessary? In some cases couldn't this cause gbench to run *more* iterations than it would automatically?

Reply:
Can you try with and without fixed iterations? It's possible that it's not the culprit and all time is spent on data generation.

Reply:
For all size except 100M, iterations > 1000 (typically 20,000). For 100M size too, it's usually >500 iterations.
The error % is 1-2% for automatic vs 1000 iterations.

Reply:
when I run automatically, it takes **6m46s**. If this is acceptable, 1000 hardcoding can be removed. (I am not fan of hardcoding either. I explored a bit for limiting maximum iterations. Right now, it's not possible with google benchmark). 
Is this acceptable?

Reply:
For automatic iterations, here is some data (Row labels is  _input size_)
![image](https://user-images.githubusercontent.com/6488848/88324308-2412c680-cd41-11ea-8de2-7fd7a9c6bdb2.png)

When fixing number of iterations of 10k,100k, 1M to 10,000 and 10M, 100M as automatic, it takes **4m21s.**

Reply:
Removed the hardcoding.
Note: It's possible to limit iterations w.r.t to input size.

Reply:
FWIW this is where the number of iterations is computed: https://github.com/google/benchmark/blob/b874e72208b6e21b62287942e5e3b11f6630107f/src/benchmark_runner.cc#L252-L273

I have a suggestion. Since gbenchmark doesn't have a command line option to specify the number of iterations, let's add our own. We may need to put this in a place that we can inherit it in all cudf benchmarks, but this would solve this problem: When running locally we can just run with auto-determined iterations to get accurate results. When running in CI, we can limit the number of iterations to a maximum using the command line, to reduce CI run time.

Thoughts?

Reply:
Really it would be better to contribute this as an improvement back to google benchmark...

Reply:
google benchmark had `--benchmark_iterations` argument and it was removed. It is replaced with `--benchmark_min_time` in this PR https://github.com/google/benchmark/pull/105

when we run on CI, we can use `--benchmark_min_time` option. In nightly, we can have bit more relaxed min time than CI.

Reply:
@devavret with 1 iteration, reduction benchmark takes 1 minute to run. So, I am guessing, data generation is not the major overhead. (out of 6m46s, 1 minute is spent on data generation).

Reply:
@harrism `--benchmark_min_time` still poses an issue when a benchmark takes very little time and measured without much jitter hence can be measured with lesser iterations. But this benchmark will still be run for that min_time.

Reply:
We want a max time, not a min time!